### PR TITLE
[SYCL][COMPAT] Add default ctor to dim3 and update tests/docs

### PR DESCRIPTION
**Pull request by joeatodd**
This PR adds a default constructor for `syclcompat::dim3`, and changes the type of members `x`,`y`,`z` from `const size_t` to `unsigned int`. The default constructor sets all 3 dimensions to `1`. This means patterns like this are now possible:

```cpp
syclcompat::dim3 myDim3;
myDim3.x = 32;
```

### CODE REVIEW

**Comment by joeatodd** [comment]
@Alcpz the request from the DPCT team also suggested changing from `size_t` to `unsigned int`. I was hesitant since `size_t` matches `sycl::range` members, but realistically I'm not sure how much of a problem this is. I suppose I ought to `static_cast<unsigned int>` in the `dim3` constructors taking a `sycl::range` argument?

File: sycl/doc/syclcompat/README.md [hunk 1]
```diff
@@ -106,9 +112,7 @@ inline dim3 operator-(const dim3 &a, const dim3 &b);
 } // syclcompat
 ```
 
-In SYCL, the fastest-moving dimension is the one with the highest index, e.g. in
-a SYCL 2D range iteration space, there are two dimensions, 0 and 1, and 1 will
-be the one that "moves faster". The compatibility headers for SYCL offer a
+The compatibility headers for SYCL offer a
```
Comment (tangjj11) [inline sycl/doc/syclcompat/README.md:115]: Does this line need format? It is too short.

**Review by tangjj11 — APPROVED** [review]

**Review by Alcpz — APPROVED** [review]

File: sycl/doc/syclcompat/README.md (hunk 1 above)
Comment (joeatodd) [reply sycl/doc/syclcompat/README.md]: Yes, well spotted - thanks :+1:

**Comment by joeatodd** [comment]
@intel/llvm-gatekeepers this is ready to merge - failures appear unrelated.

**Comment by steffenlarsen** [comment]
@joeatodd - The description reads a little like a conversation, so I am a little reluctant to put it as a commit message. Would you mind rewording it and include the information about the change in parameters?

**Comment by joeatodd** [comment]
@steffenlarsen thanks for the suggestion - done :+1:

**Pull request merged**